src/systemctl/systemctl.c

```diff
@@ -6341,8 +6341,27 @@ static int unit_is_enabled(int argc, char *argv[], void *userdata) {
                         if (STR_IN_SET(s, "enabled", "enabled-runtime", "static", "indirect", "generated"))
                                 enabled = true;
 
-                        if (!arg_quiet)
+                        if (!arg_quiet) {
                                 puts(s);
+                                if (arg_full) {
+                                        UnitFileFlags flags = UNIT_FILE_DRY_RUN;
+                                        UnitFileChange *changes = NULL;
+                                        unsigned n_changes = 0, i;
+                                        _cleanup_strv_free_ char **p = NULL;
+
+                                        flags |= arg_runtime & UNIT_FILE_RUNTIME;
+                                        p = strv_new(*name, NULL);
+                                        if (!p)
+                                                return log_oom();
+
+                                        r = unit_file_disable(UNIT_FILE_SYSTEM, flags, NULL, p, &changes, &n_changes);
+                                        if (r < 0)
+                                                return log_error_errno(r, "Failed to get targets for %s: %m", *name);
```

Comment:
Hmm, so now hooked this into the codepath when systemctl talks to PID 1 for enablement/disablement commands, but for your code you actually look at the unit files directly. This is strange of course. I figure you should add a new bus call for this and call that here. GetUnitLinks() or something like this, and call that here...